testing/e2e/app/routes/Select/spec.ts

```diff
@@ -0,0 +1,143 @@
+        }
+
+        await expect(page.getByRole('combobox')).toHaveText('option 1');
+      });
+    });
+  });
+
+  test.describe('Overflow', () => {
+    test(`should overflow whenever there is not enough space`, async ({
+      page,
+    }) => {
+      await page.goto(`/Select?multiple=true&value=all`);
+
+      const setContainerSize = getSetContainerSize(page);
+      const expectOverflowState = getExpectOverflowState(page);
+
+      await page.waitForTimeout(200);
+
+      await expectOverflowState({
+        expectedItemLength: 10,
+        expectedLastTagTextContent: 'Very long option',
+      });
+
+      await setContainerSize('600px');
+
+      await expectOverflowState({
+        expectedItemLength: 6,
+        expectedLastTagTextContent: '+5 item(s)',
+      });
+    });
+
+    test(`should at minimum always show one overflow tag`, async ({ page }) => {
+      await page.goto(`/Select?multiple=true&value=all`);
+
+      const setContainerSize = getSetContainerSize(page);
+      const expectOverflowState = getExpectOverflowState(page);
+
+      await page.waitForTimeout(200);
+
+      await expectOverflowState({
+        expectedItemLength: 10,
+        expectedLastTagTextContent: 'Very long option',
+      });
+
+      await setContainerSize('10px');
+
+      await expectOverflowState({
+        expectedItemLength: 1,
+        expectedLastTagTextContent: '+10 item(s)',
+      });
+    });
+
+    test(`should always show the selected tag and no overflow tag when only one item is selected`, async ({
+      page,
+    }) => {
+      await page.goto(`/Select?multiple=true&value=[9]`);
+
+      const setContainerSize = getSetContainerSize(page);
+      const expectOverflowState = getExpectOverflowState(page);
+
+      await page.waitForTimeout(200);
+
+      await expectOverflowState({
+        expectedItemLength: 1,
+        expectedLastTagTextContent: 'Very long option',
+      });
+
+      await setContainerSize('160px');
+
+      await expectOverflowState({
+        expectedItemLength: 1,
+        expectedLastTagTextContent: 'Very long option',
+      });
+    });
+  });
+});
+
+// ----------------------------------------------------------------------------
+
+const getSetContainerSize = (page: Page) => {
+  return async (dimension: string | undefined) => {
+    await page.locator('#container').evaluate(
+      (element, args) => {
+        if (args.dimension != null) {
+          element.style.setProperty('width', args.dimension);
+        } else {
+          element.style.removeProperty('width');
+        }
+      },
+      { dimension },
+    );
+    await page.waitForTimeout(200);
+  };
+};
+
+const getExpectOverflowState = (page: Page) => {
+  return async ({
+    expectedItemLength,
+    expectedLastTagTextContent,
+  }: {
+    expectedItemLength: number;
+    expectedLastTagTextContent: string | undefined;
+  }) => {
+    const tags = await page
+      .locator('div[class$="-select-tag-container"] > span')
```

Comment:
this locator is also relying on an implementation detail?

Reply:
Same as https://github.com/iTwin/iTwinUI/pull/2151#discussion_r1684394649, this is still relying on an implementation detail.